File: src/main/java/commands/slashadapters/adapters/BooruPredeterminedAdapterAbstract.java
package commands.slashadapters.adapters;

import commands.slashadapters.SlashAdapter;
import commands.slashadapters.SlashMeta;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.CommandData;

public abstract class BooruPredeterminedAdapterAbstract extends SlashAdapter {

    public CommandData addOptions(CommandData commandData) {
        return commandData
                .addOption(OptionType.INTEGER, "amount", "Amount of posts from 1 to 20 / 30", false);
    }

    @Override
    public SlashMeta process(SlashCommandEvent event) {
        return new SlashMeta(commandClass(), collectArgs(event));
    }

}

File: src/main/java/commands/slashadapters/adapters/BooruPredeterminedAdapterAbstract.kt
package commands.slashadapters.adapters

import commands.Category
import commands.slashadapters.SlashAdapter
import commands.slashadapters.SlashMeta
import mysql.hibernate.entity.guild.GuildEntity
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData

abstract class BooruPredeterminedAdapterAbstract : SlashAdapter() {

    public override fun addOptions(commandData: SlashCommandData): SlashCommandData {
        return commandData
            .addOptions(
                generateOptionData(OptionType.INTEGER, "amount", Category.NSFW.id, "porn_amount", false)
            )
    }

    override fun process(event: SlashCommandInteractionEvent, guildEntity: GuildEntity): SlashMeta {
        return SlashMeta(commandClass().java, collectArgs(event))
    }

}